site/src/components/Dialogs/DeleteDialog/DeleteDialog.test.tsx

```diff
@@ -2,6 +2,23 @@ import { screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 import { render } from "testHelpers/renderHelpers";
 import { DeleteDialog } from "./DeleteDialog";
+import { act } from "react-dom/test-utils";
+
+const inputTestId = "delete-dialog-name-confirmation";
+
+async function fillInputField(inputElement: HTMLElement, text: string) {
+  // 2023-10-06 - There's something wonky with MUI's ConfirmDialog that causes
+  // its state to update after a typing event gets  fired, and React Testing
+  // Library isn't able to catch it, making React DOM freak out because an
+  // "unexpected" state change happened. It won't fail the test, but it makes
+  // the console look really scary because it'll spit out a big warning message.
+  // Tried everything under the sun to catch the state changes the proper way,
+  // but the only way to get around it for now might be to manually make React
+  // DOM aware of the changes
+
+  // eslint-disable-next-line testing-library/no-unnecessary-act -- have to make sure state updates don't slip through cracks
+  return act(() => userEvent.type(inputElement, text));
```

Comment:
To be clear, this issue existed with the previous version of the component, too – it's just that the test never dealt with the warning messages getting spat out

Going to see if it's worth opening a bug report

Reply:
The logic in the tests themselves doesn't change

Reply:
Maybe the first reasonable use case for `act` I've seen in my career :)